### PR TITLE
Support remote access to devices from Administration context

### PR DESCRIPTION
**Pull request by saimonation**
- Add base_object_ref parser
- Create common/utils module
- Move union to utils module
- Update modules
- Update UT

### CODE REVIEW

**Comment by saimonation** [comment]
@ygalblum - Previously, we computed the remote access URL from the 'tenant' attribute in the user's session. This caused an issue when trying to remote access from the Administration context. The fix will parse the Portal attribute from the device object.

File: cterasdk/common/utils.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        Returns a string representation of the base object reference
+        """
+        return '/'.join(['objs', self.uid, self.tenant, self.classname, self.name])
+
+
+def parse_base_object_ref(base_object_ref):
+    """
+    Parse a base object reference.
+
+    :return cterasdk.common.utils.BaseObjectRef param: Object holding the classname, uid, tenant and object name
```
Comment (ygalblum) [inline cterasdk/common/utils.py:54]: Move param above return

Comment (saimonation) [reply cterasdk/common/utils.py]: Hi Ygal, I believe you mean add a 'param' entry to the documentation. Have a look at the latest commit

**Review by ygalblum — APPROVED** [review]

File: cterasdk/common/utils.py (hunk 1 above)
Comment (ygalblum) [reply cterasdk/common/utils.py]: Yes that's what I meant :)

**Pull request merged**